File: tusow/src/main/java/it/unibo/coordination/tusow/exceptions/InternalServerError.java
package it.unibo.coordination.tusow.exceptions;

public class InternalServerError extends ServerError {

    private static final int CODE = 500;

    private static final String MSG = "Internal Server Error";

    public InternalServerError() {
        super(CODE, MSG);
    }

    public InternalServerError(String message) {
        super(CODE, MSG + ": " + message);
    }

    public InternalServerError(String message, Throwable cause) {
        super(CODE, MSG + ": " +  message, cause);
    }

    public InternalServerError(Throwable cause) {
        super(CODE, MSG, cause);
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/exceptions/InternalServerError.kt
package it.unibo.coordination.tusow.exceptions

class InternalServerError : ServerError {
    constructor() : super(CODE, MSG)
    constructor(message: String) : super(CODE, "$MSG: $message")
    constructor(message: String, cause: Throwable?) : super(CODE, "$MSG: $message", cause)
    constructor(cause: Throwable?) : super(CODE, MSG, cause)

    companion object {
        private const val CODE = 500
        private const val MSG = "Internal Server Error"
    }
}